### PR TITLE
Update Browser e2e tests to use MazeRunner v4

### PR DESCRIPTION
**Pull request by twometresteve**
## Goal

Update the Browser tests to use Maze Runner v4.

## Design

MazeRunner v4 accepts errors and sessions on separate endpoints and also absorbs the Selenium driver code previously present in the `bugsnag-js` repository.  This, combined with the changes in MazeRunner v3 that provide better support for running outside of a Docker container, mean that the method of running tests locally can be revised to be simpler, quicker, and more consistent with our other notifier repositories.

## Changeset

This change touches a lot of files, but you will see that any of the differences are highly repetitive.

- Test fixtures configured to send to separate endpoints
- Cucumber steps updated for v4 changes
- Selenium driver code removed
- Pipeline steps updated to use v4 options
- Mechanism and instructions for running locally updated
 
## Testing

Covered by CI.

### CODE REVIEW

File: dockerfiles/Dockerfile.browser [hunk 1]
```diff
@@ -45,7 +45,8 @@ RUN find . -name package.json -type f -mindepth 2 -maxdepth 3 ! -path "./node_mo
 RUN rm -fr **/*/node_modules/
 
 # The maze-runner browser tests
-FROM 855461928731.dkr.ecr.us-west-1.amazonaws.com/maze-runner-releases:latest-v3-cli as browser-maze-runner
+FROM 855461928731.dkr.ecr.us-west-1.amazonaws.com/maze-runner:tms-browser-corrections-cli as browser-maze-runner
+#FROM 855461928731.dkr.ecr.us-west-1.amazonaws.com/maze-runner-releases:latest-v4-cli as browser-maze-runner
```
Comment (twometresteve) [inline dockerfiles/Dockerfile.browser:49]: Some corrections to MazeRunner were needed to support this change and are awaiting review. Once merged, MR can be re-released and this simply changed back.

File: test/browser/Gemfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+source 'https://rubygems.org'
```
Comment (imjoehaines) [inline test/browser/Gemfile:1]: Does this exist so the browser tests can use a different MR version to the rest of the tests?

File: test/browser/Gemfile [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+gem 'cocoapods'
```
Comment (imjoehaines) [inline test/browser/Gemfile:3]: I don't think this needs cocoapods — isn't that just for the RN CLI tests?

File: test/browser/Gemfile [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+# A reference to Maze Runner is only needed for running tests locally and if committed it must be
+# portable for CI, e.g. a specific release.  However, leaving it commented out would mean quicker CI.
```
Comment (imjoehaines) [inline test/browser/Gemfile:6]: This comment doesn't apply if we're deleting this file in the dockerfile

**Review by imjoehaines — APPROVED** [review]

File: test/browser/Gemfile (hunk 2 above)
Comment (twometresteve) [reply test/browser/Gemfile:1]: Yes, exactly, and adding it caused some issues with the Docker image, hence the removal above - although I'm not sure that was actually necessary now given that it builds without deleting what I intended it to.

File: test/browser/Gemfile (hunk 3 above)
Comment (twometresteve) [reply test/browser/Gemfile]: Yep, copy/paste error.

File: dockerfiles/Dockerfile.browser [hunk 5]
```diff
@@ -23,6 +23,7 @@ RUN npm pack --verbose packages/plugin-react/
 RUN npm pack --verbose packages/plugin-vue/
 
 COPY test/browser test/browser
+RUN rm test/browser Gemfile*
```
[the remark this answers is not in the document]
Comment (twometresteve) [reply dockerfiles/Dockerfile.browser]: Corrected.

**Pull request merged**